src/QueryBuilder.php

```diff
@@ -27,276 +26,448 @@ class QueryBuilder extends Object {
 
     /**
      * Constructor.
-     * @param Connection $connection the database connection.
-     * @param array $config name-value pairs that will be used to initialize the object properties
+     * @param Connection $connection The database connection.
+     * @param array      $config     Name-value pairs that will be used to initialize the object properties.
      */
-    public function __construct(Connection $connection, $config = []) {
+    public function __construct(Connection $connection, array $config = [])
+    {
         $this->db = $connection;
         parent::__construct($config);
     }
-    
+
     /**
      * Generates DynamoDB Query from a [[Query]] object.
-     * @param Query $query object from which the query will be generated.
-     * @return array the generated DynamoDB command configuration
+     * @param Query $query Object from which the query will be generated.
+     * @return array The generated DynamoDB command configuration.
      */
-    public function build(Query $query) {
-        if ($query->using == Query::TYPE_BATCH_GET) {
-            $request = $this->buildBatchGetItem($query);
-        } else if ($query->using == Query::TYPE_GET) {
-            $request = $this->buildGetItem($query);
+    public function build(Query $query)
+    {
+        // Validate query
+        if (empty($query->from)) {
+            throw new \InvalidArgumentException('Table name not set');
+        }
+        if ($query->using == Query::USING_AUTO) {
+            if(!empty($query->where)) {
+                // TODO Choose either SCAN OR QUERY
+                $query->using = Query::USING_SCAN;
+            } else {
+                // TODO Look at keys
+                throw new \Exception('Unsupported query condition');
+            }
         }
-        return [
-            'method'=> $query->using,
-            'request' => $request
-        ];        
-    }
 
-    private function buildGetItem(Query $query) {
-        $request = [];
-        // required
-        $request['Key'] = $this->buildKey($query);
-        $request['TableName'] = $this->buildTableName($query);
+        $call = 'build' . $query->using;
 
-        // optional
-        $this->setCommonRequest($request, $query);
-        $this->buildConsumedCapacity($request, $query);
-        return $request;
-    }
-    private function buildConsumedCapacity(&$request, &$query) {
-        if (isset($query->returnConsumedCapacity)) {
-            $request['ReturnConsumedCapacity'] = $query->returnConsumedCapacity;
-        }
+        // Call builder
+        return $this->$call($query);
```

Comment:
Actually I was going to recommend using `call_user_func` but actually this is faster.
Anyway, use braces for the dynamic function or field call like this. Much readable.

```
$this->{$call}($query);
```